src/boosting/gbdt.cpp

```diff
@@ -17,6 +17,8 @@
 
 namespace LightGBM {
 
+Common::Timer global_timer;
```

Comment:
moving this define from `application.cpp` to this file (the first source file that needs `global_timer`) to try to fix these errors:

https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=8009&view=logs&j=5ea309bb-dea2-5bcb-538b-5c76ecf159eb&t=aa1d7e0f-3509-5697-4ef8-f247c54ed420